### PR TITLE
docs(README) update automation guide

### PR DESCRIPTION
**Pull request by kikito**

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -75,30 +75,39 @@ The Enterprise documentation uses paid algolia indices, which auto-update every
...
-      `autodoc-admin-api/data.lua` and error out if there's any mismatches or missing data.
-    * If no errors were found, a new `admin-api.md` file will be generated in the path corresponding
-      to the provided KONG_VERSION.
+  - On the kong/kong repo, run `./scripts/autodoc-admin-api`
+  - Copy the output from `kong/kong/autodoc/output/admin-api/` into `kong/docs/app/0.14.x` (replace with current release)
+  - Copy `kong/kong/autodoc/output/nav/docs_nav.yml.admin-api.in` into `kong/docs/autodoc-nav/docs_nav_0.14.x.yml.head.in`. Replace 0.14.x with release.
+  - On the kong/docs repo, run `KONG_VERSION=0.14.x luajit ./autodoc-nav/run.lua` (replace `0.14.x` with current release). This will merge the navigation
+    info in the right place.
+  - These commands generate a big file for the admin API and a smaller file for the navigation, which needs to be inserted in a
+  - specific place in the navigation yaml file.
```
Comment (hishamhm) [inline README.md:94]: @kikito The smaller file is used as input by the `autodoc-nav` script. And it also generates another big Admin API file specific for DB-less.

File: README.md [hunk 2]
```diff
@@ -75,30 +75,39 @@ The Enterprise documentation uses paid algolia indices, which auto-update every
...
 - To generate the Configuration docs:
-  - Run: `KONG_PATH=path/to/your/kong/folder KONG_VERSION=0.14.x gulp conf-docs`
+  - In kong/docs: `KONG_PATH=path/to/your/kong/folder KONG_VERSION=0.14.x gulp conf-docs` (replace 0.14.x with current release)
   - This command will:
     * Parse Kong's `kong.conf.default` file and extract sections, variable names, descriptions, and default values
     * Write those down inside a `configuration.md` file in the path matching KONG_VERSION.
     * The command will completely overwrite the file, including text before and after the list of vars.
     * The data used for the before/after parts can be found in `autodoc-conf/data.lua`
 
+- Once everything is generated, git add, git push, and review.
```
Comment (hishamhm) [inline README.md:109]:
```suggestion
- Once everything is generated, submit the generated changes as a pull request.
```

I think it's better to either list the actual commands exactly as need to be executed, or assume the reader knows how to open a pull request (this sits halfway: "git add, git push" look like commands to run but are shorthands that assume that the reader knows how to open a branch, push to that branch, etc)

File: README.md [hunk 3]
```diff
@@ -75,30 +75,39 @@ The Enterprise documentation uses paid algolia indices, which auto-update every
...
+
 - To generate the CLI docs:
-  - Run: `KONG_PATH=path/to/your/kong/folder KONG_VERSION=0.14.x gulp cli-docs`
+  - In kong/docs `KONG_PATH=path/to/your/kong/repo KONG_VERSION=0.14.x gulp cli-docs` (replace 0.14.x with current release)
   - This command will:
     * Extract the output of the `--help` for every `kong` CLI subcommand
     * Generate a new `cli.md` in the path corresponding to the provided KONG_VERSION.
 - To generate the Configuration docs:
-  - Run: `KONG_PATH=path/to/your/kong/folder KONG_VERSION=0.14.x gulp conf-docs`
+  - In kong/docs: `KONG_PATH=path/to/your/kong/folder KONG_VERSION=0.14.x gulp conf-docs` (replace 0.14.x with current release)
```
Comment (hishamhm) [inline README.md:102]: Not something to change here but a heads-up: none of the automation that is currently in place for running the generation scripts at release time use `gulp` and the associated JavaScript plumbing. If any changes are made to the process using `gulp`, it won't be picked up there.

Ideally, what should happen is that a "docs generation script" should be extracted out of the "Kong release script" and used both there and here, which would turn the whole process into a one-liner.

**Comment by kikito** [comment]
Thanks for the feedback, I made the following changes to reflect hisham's feedback:
* Added a note about the Core team using a different method
* Added a link to our script
* Removed the references to gulp when alternatives existed.
* Added a note in the PDK section pointing out that we don't use gulp for that task.

**Pull request merged**